Dockerfile.mono

```diff
@@ -4,15 +4,21 @@ ARG mono_version
 
 RUN if [ -z "${mono_version}" ]; then echo -e "\n\nargument mono-version is mandatory!\n\n"; exit 1; fi
 
-RUN dnf -y install cmake gcc gcc-c++ make which perl python curl bzip2 && dnf clean all && \
-    curl https://download.mono-project.com/sources/mono/mono-${mono_version}.tar.bz2 | tar xj && \
+RUN dnf -y install autoconf automake libtool cmake gcc gcc-c++ make which perl python curl bzip2 xz && dnf clean all && \
+    if [ ${mono_version%%.*} -ge 6 ]; then \
```

Comment:
As mentioned on https://github.com/godotengine/build-containers/pull/7#issuecomment-536419525, I had to use a Git clone for the Android build, so I might as well do the same for all the builds here?
That also removes the awkwardness of handling the switch from bzip to xz in 6.0 :P